src/NUnitFramework/framework/Internal/Execution/CommandBuilder.cs

```diff
@@ -57,6 +57,12 @@ public static TestCommand MakeOneTimeSetUpCommand(TestSuite suite, List<SetUpTea
                 if (changes.Length > 0)
                     command = new ApplyChangesToContextCommand(command, changes);
             }
+            if (suite is ParameterizedMethodSuite && suite.HasChildren)
+            {
+                IApplyToContext[] changes = (IApplyToContext[])suite.Tests[0].Method.GetCustomAttributes(typeof(IApplyToContext), true);
+                if (changes.Length > 0)
+                    command = new ApplyChangesToContextCommand(command, changes);
+            }
```

Comment:
Good! You can simplify this by eliminating the HasChildren test and using suite.Method, which is inherited from Test.

Thinking about it you may be able to just test if (suite.Method != null) so it would work for any future suite built around a method rather than a type.